### ISSUE TITLE
Spells

### ISSUE THREAD

**Issue by sansthesis** [chronicles-mud/chronicles#11]
Do we keep the list of stock spells?  Do we implement other ones that we like?

Should spells be moved around groups?

What spells should new classes get?

**Comment by sansthesis** [comment]
I personally think that the cleric and mage spell lists are full of bloat, and that we should try to get the spell lists down to maybe 20-30 for each, with as little overlap as possible.

Mages should specialize in elemental damaging magic and clerics should specialize in divine/unholy magic, IMO.

**Comment by XTremeTennis** [comment]
Here is the ansalon mage spell list (93 spells);
Lvl  1: magic missile       50 mana (  1%)   ventriloquate       50 mana (  1%)
Lvl  3: detect magic             n/a
Lvl  5: chill touch              n/a         invisibility             n/a
Lvl  6: continual light
Lvl  7: armor
Lvl  9: burning hands      detect invis     infravision
Lvl 12: conceal alignment          shocking grasp
Lvl 13: fireproof                n/a         fly
Lvl 15: free action              n/a         identify
Lvl 16: faerie fire              n/a         lightning bolt
Lvl 18: blindness        penetrate energy      refresh
Lvl 19: colour spray             n/a         detect evil              n/a      detect good              n/a         detect neutral           n/a      locate object            n/a         weaken
Lvl 20: confusion                n/a         control weather          n/a      sleep
Lvl 22: penetrate fire
Lvl 23: teleport
Lvl 24: faerie fog               n/a         penetrate cold
Lvl 25: detect hidden            n/a         detect poison            n/a      grease
Lvl 26: giant strength
Lvl 27: penetrate acid           n/a         poison
Lvl 28: cancellation             n/a         fireball       haste                    n/a         penetrate poison
Lvl 29: energy drain
Lvl 30: charm person           corpse visage      flood waters            understand tongues
Lvl 32: know alignment           n/a         protection element
Lvl 33: slow
Lvl 34: pass door                n/a         summon
Lvl 36: call lightning                dispel magic               enchant armor               word of recall
Lvl 37: find familiar            n/a         gate
Lvl 38: acid blast
Lvl 39: enchant weapon           n/a         recharge
Lvl 40: fear
Lvl 42: mass invis               n/a         stone skin
Lvl 43: plague                   n/a         silence
Lvl 45: mass fly                 n/a         portal 
Lvl 47: shield
Lvl 48: scrye
Lvl 50: thunderclap
Lvl 53: chain lightning
Lvl 55: smoke
Lvl 57: sanctuary
Lvl 60: nexus                    n/a         prismatic spray          n/a      translate                n/a         web
Lvl 62: acid breath
Lvl 65: earthshield              n/a         iceshield
Lvl 70: protection
Lvl 74: flameshield              n/a         frost breath             n/a      lightning breath
Lvl 75: shockshield              n/a         thunderstorm
Lvl 76: true sight
Lvl 78: gas breath
Lvl 80: fire breath
Lvl 101: fire storm

I think that could at the very least be simplified to (58 spells);
Lvl  1: magic missile       50 mana (  1%) 
Lvl  5: chill touch              n/a         invisibility
Lvl  7: armor
Lvl  9: detect invis
Lvl 13: fireproof                n/a         fly
Lvl 15: free action              n/a         identify
Lvl 16: lightning bolt
Lvl 18: blindness                n/a         penetrate energy         n/a      refresh
Lvl 19: locate object            n/a         weaken
Lvl 20: confusion                n/a        sleep
Lvl 22: penetrate fire
Lvl 23: teleport
Lvl 24: penetrate cold
Lvl 26: giant strength
Lvl 27: penetrate acid           n/a         poison
Lvl 28: cancellation             n/a         fireball
        haste                    n/a         penetrate poison
Lvl 30: charm person
Lvl 32: protection element
Lvl 33: slow
Lvl 34: pass door                n/a         summon
Lvl 36: dispel magic             n/a      enchant armor            n/a         word of recall
Lvl 37: find familiar            n/a         gate
Lvl 38: acid blast
Lvl 39: enchant weapon           n/a         recharge
Lvl 42: mass invis               n/a         stone skin
Lvl 43: plague                   n/a         silence
Lvl 45: mass fly                 n/a         portal
Lvl 47: shield
Lvl 48: scrye
Lvl 53: chain lightning
Lvl 57: sanctuary
Lvl 60: nexus                    n/a     web
Lvl 65: earthshield              n/a         iceshield
Lvl 70: protection
Lvl 74: flameshield
Lvl 75: shockshield
Lvl 76: true sight
Lvl 101: fire storm

That mage spell list is still large and clunky, but is cleared out of a lot of crap. The way I see it mages should have damage spells that work like this;

one mob spells: Magic Missile, Fireball, Lightning Bolt, Acid Arrow, Ray of Frost
AoE spells: Blizzard, Firestorm, Chain Lightning, Acid Cloud, Prismatic Spray
with the possibility for a "penetrate" type spell for each damage type, or maybe even a universal penetrate type spell.

Mage buffs should be more limited, I like the elemental shield ideas, and then maybe self casting buffs? Mage Armor and Shield that can only be cast on self? Changing the buff spells so that mages and clerics aren't almost identical is important.

Debuffs; should clerics and mages have different debuffs?

Utility spells: Fly and Invisibility seem pretty standard, as with the mass versions of those spells. At one point removing gate and replacing with portal was talked about, mostly so that melee classes could follow a mage during PK, that is something to consider at least. It would be cool if enchant armor and enchant weapon were actually things rather than useless.

**Comment by XTremeTennis** [comment]
Also, do players really need to lose the ability to recall? I don't honestly see the harm in just having recall be a command for everyone all the time.

**Comment by sansthesis** [comment]
I'd like to strip out more.

I think that armor and shield are redundant.  One should be buffed and the other should be removed.

Penetration should just do all the elemental resistances for mages and be merged into one spell for them, then renamed and added to the maledictions group for them.

It'd be helpful to bucket them like you started doing:

| Category | Arcane(?) | Fire | Frost | Electric | Acid |
| --- | --- | --- | --- | --- | --- |
| Single Target | Magic Missile | Burning Hands | Ray of Frost | Shocking Grasp | Acid Blast |
| AOE | Prismatic Spray | Firestorm | Blizzard | Thunderstorm | Horrid Wilting |

etc

**Comment by XTremeTennis** [comment]
I wonder about maledictions too, I think some make more sense than others for mages, plague on a mage always seemed weird to me.

Combat Spells:

| Category | Arcane/Energy | Fire | Frost | Electric | Acid |
| --- | --- | --- | --- | --- | --- |
| Single Target | Magic Missile | Burning Hands | Ray of Frost | Shocking Grasp (I like lightning bolt more) | Acid Blast |
| AOE | Prismatic Spray | Firestorm | Blizzard | Thunderstorm | Horrid Wilting |

Maladicts/Debuffs:

| Yes | Maybe |
| --- | --- |
| Sleep | Poison |
| Silence | Plague |
| Slow | Weaken |
| Penetrate X | Confusion |
| Dispel | Blindness |

Buffs:

| Yes | Maybe | No way |
| --- | --- | --- |
| Haste | Stone Skin | Sanctuary |
| Invis | Protection | Armor |
| Fly | Giant Str | X |
| Shield | Pass Door | X |
| Elemental Shield (ice, frost, earth, etc) | Protection Element | X |
| Detect Invis | True Sight | X |
| X | Free Action | X |

Utility Spells:
Gate and/or Portal
Summon
Cancellation (although if players could just willingly de-buff themselves that would work too.)
Locate Item (If we switch to bulk random eq, then this isn't needed)
Scrye
Enchant Weapon/Armor
Refresh?
Find Familiar
Charm
Teleport

**Comment by sansthesis** [comment]
I don't think that the AOE spells should have any duration other than instant, so I'm trying to find an acid type spell that would work with that restriction

| Category | Arcane/Energy | Fire | Frost | Electric | Acid |
| --- | --- | --- | --- | --- | --- |
| Single Target | Magic Missile | Burning Hands | Ray of Frost | Lightning Bolt | Acid Blast |
| AOE | Prismatic Spray | Fireball | Cone of Cold | Chain Lightning | Horrid Wilting |

**Comment by XTremeTennis** [comment]
Could always just make one up, even just say acid breath and say its a cone

**Comment by sansthesis** [comment]
Yeah, I could use acid breath, but I don't want the two spells to have similar names, so something else for the single target then?

**Comment by XTremeTennis** [comment]
It's not really an acid spell, but cloudkill is a good name and not like any of the other spells

**Comment by XTremeTennis** [comment]
This is the cleric spell list (102 spells);
- Lvl  1: cause light         50 mana    cure light          50 mana  slow poison         50 mana 
- Lvl  2: armor                    n/a  
- Lvl  4: continual light          n/a  
- Lvl  6: refresh                  n/a  
- Lvl  7: bless                    n/a  
- Lvl  8: cure serious             n/a         detect evil              n/a      detect good              n/a         detect neutral           n/a  
- Lvl  9: detect poison            n/a  
- Lvl 10: cause serious            n/a         icicle                   n/a  
- Lvl 12: blindness                n/a         fireproof                n/a      poison                   n/a  
- Lvl 13: faerie fire              n/a         infravision              n/a  
- Lvl 14: weaken                   n/a  
- Lvl 15: conceal alignment        n/a         know alignment           n/a  
- Lvl 16: cure blindness           n/a         cure poison              n/a      identify                 n/a  
- Lvl 17: cure disease             n/a         invisibility             n/a  
-  18: call lightning           n/a         confusion                n/a      cure critical            n/a         detect invis             n/a  
- Lvl 19: protection neutral       n/a         protection evil          n/a      protection good          n/a  
- Lvl 21: cause critical           n/a  
- Lvl 23: harm                     n/a         lightning bolt           n/a  
- Lvl 24: control weather          n/a         corpse visage            n/a      fly                      n/a         frenzy                   n/a      heal                     n/a  
- Lvl 25: dispel evil              n/a         dispel good              n/a      dispel neutral           n/a         locate object            n/a  
- Lvl 26: detect magic             n/a  
-  Lvl 27: plague                   n/a  
- Lvl 28: curse                    n/a         remove curse             n/a  
- Lvl 30: shield                   n/a  
- Lvl 31: detect hidden            n/a         faerie fog               n/a  
- Lvl 32: clay spire               n/a         understand tongues       n/a  
- Lvl 34: demonfire                n/a  
- Lvl 35: free action              n/a         stone skin               n/a  
- Lvl 36: cancellation             n/a         heat metal               n/a  
- Lvl 38: fear                     n/a  
- Lvl 39: giant strength           n/a  
- Lvl 40: charm person             n/a         flamestrike              n/a      sanctuary                n/a         slow 
- Lvl 42: haste                    n/a         teleport                 n/a  
- Lvl 43: mass healing             n/a  
- Lvl 44: dispel magic             n/a         word of recall           n/a  
- Lvl 45: earthshield              n/a         flood waters             n/a      ray of truth             n/a  
- Lvl 46: calm                     n/a  
- Lvl 47: gate                     n/a  
- Lvl 48: silence                  n/a  
- Lvl 52: aqua pillar              n/a         pass door                n/a      summon                   n/a  
- Lvl 53: mass fly                 n/a  
- Lvl 55: mass invis               n/a         protection               n/a  
- Lvl 56: holy word                n/a  
- Lvl 60: charge water             n/a  
- Lvl 62: scrye                    n/a         translate                n/a  
- Lvl 65: earthen spire            n/a         thunderclap              n/a  
- Lvl 70: portal                   n/a  
- Lvl 72: iceshield                n/a  
- Lvl 75: shockshield              n/a  
- Lvl 80: entangle                 n/a         flameshield              n/a  
- Lvl 85: crushing wave            n/a         nexus                    n/a      smoke                    n/a         thunderstorm             n/a  
- Lvl 87: true sight               n/a  
- Lvl 90: earthquake               n/a         holy smite               n/a       protection element       n/a  

And this is one with the easy cuts made (66 spells):
- Lvl  2: armor                    n/a  
- Lvl  6: refresh                  n/
- Lvl  7: bless                    n/
- Lvl  8: detect evil              n/a       detect good              n/a         detect neutral           n/
- Lvl 12: blindness                n/a         fireproof                n/a      poison                   n/
- Lvl 14: weaken                   n/
- Lvl 15: conceal alignment        n/a         know alignment           n/
- Lvl 16: cure blindness           n/a         cure poison              n/a      identify                 n/
- Lvl 17: cure disease             n/a         invisibility             n/
- Lvl 18: confusion                n/a       detect invis             n/
- Lvl 19: protection neutral       n/a         protection evil          n/a      protection good          n/
- Lvl 23: harm                     n/a
- Lvl 24: fly                      n/a         frenzy                   n/a      heal                     n/a  
- Lvl 25: dispel evil              n/a         dispel good              n/a      dispel neutral           n/a         locate object            n/
- Lvl 27: plague                   n/
- Lvl 28: curse                    n/a         remove curse             n/
- Lvl 30: shield                   n/
- Lvl 34: demonfire                n/
- Lvl 35: free action              n/a         stone skin               n/
- Lvl 36: cancellation             n/
- Lvl 38: fear                     n/
- Lvl 39: giant strength           n/
- Lvl 40: charm person             n/a           sanctuary                n/a         slow                     n/
- Lvl 42: haste                    n/a         teleport                 n/
- Lvl 43: mass healing             n/
- Lvl 44: dispel magic             n/a         word of recall           n/
- Lvl 45: ray of truth             n/
- Lvl 46: calm                     n/
- Lvl 47: gate                     n/
- Lvl 48: silence                  n/
- Lvl 52: pass door                n/a      summon                   n/
- Lvl 53: mass fly                 n/a  
- Lvl 55: mass invis               n/a         protection               n/
- Lvl 56: holy word                n/
- Lvl 62: scrye                    n/
- Lvl 70: portal                   n/
- Lvl 80: entangle                 n/
- Lvl 85: nexus                    n/a        thunderstorm             n/
- Lvl 87: true sight               n/
- Lvl 90: holy smite               n/a      protection element       n/

**Comment by XTremeTennis** [comment]
For clerics there is also some easy reduction possible by making the dispel good/evil/neutral just a spell called "dispel" that affects everyone of a different align than you, same goes for protection good/evil/neutral. 
You could combine all the remove curse/cure disease/cure poison/cure blindness into a single spell as well, but I don't know if you would want to.

**Comment by XTremeTennis** [comment]
I see clerics working like this:

Divine Damage Spells;

| Target | Spell | Notes |
| --- | --- | --- |
| Single | Harm | Pure damage to single target, maybe magic in nature? |
| Single | Dispel(?) | Damage affected by alignment, can do more than harm to outside of align but less inside align? |
| AoE | Wrath | AoE alignment based spell |

Healing:

| Spell | Note |
| --- | --- |
| Cure | Obtained at level 1, heals in a range that gets bigger the higher in level you are |
| Heal | Obtained later, costs more mana than cure but heals an assigned amount every time |
| Refresh | Heals moves |

Buffs:

| Spell | Notes |
| --- | --- |
| Bless | Bonus to hitroll and maybe saves |
| Armor | bonus to Ac or DR or both |
| Sanctuary | Damage reduction |
| Divine Blessing | I think an alignment based buff spell would be cool, where it works different based on what the casters align is |
| Protection | less damage from those outside alignment, then we can remove protection X alignment |

Debuffs:

| Spell | Notes |
| --- | --- |
| Calm | makes target unable to attack |
| Blindness | blinds someone |
| Weaken | lowers strength/damage roll |
| Curse | lowers hitroll |
| Plague | I'm not sure if we actually need plague |
| Poison | damage over time |

Utility Spells
- Remove/Cure X 
- Gate/Portal
- Summon
- Detect Align/Concel align (?) - Seems like it has potential to be interesting if you make clerics more alignment based, but could also be stupid
- Fly
- Invis

**Comment by sansthesis** [comment]
I think we roll stone skin into armor at higher levels, like armor provides ac at lower levels, then at a certain point provides armor and weapon resistance.

And maybe do the same with shield and the protection element spells for mages.

And give mages shield and clerics armor.

**Comment by XTremeTennis** [comment]
Chris had talked about implementing straight damage resistance at some point, but improving spells is a decent idea, I like the idea of shield instead of providing weapon or DR providing energy resist, and earthshield providing acid resist so that mages are more about protection from elements than physical damage. More focus on elemental damage types and resistances on mobs would make it so you might have to spell up differently in different areas and use different spells so there is more of a reason for these damtypes

**Comment by sansthesis** [comment]
I was thinking of removing the elemental shield spells entirely and folding their resistances into shield.  I'd add energy resistance to shield so that it affects the 5 elemental types that we're setting mages up to have mastery over.

I'd do the same with stone skin by rolling in weapon resistance, holy, and unholy resistance into armor for clerics, so that clerics and mages have a clear bifurcation of their domains.

**Comment by sansthesis** [comment]
This would also allow us to remove sanctuary as a spell, since we'll be handing out all kinds of damage reduction left and right.

I'm currently concerned that the mage defensive buffs really only affect mage<->mage combat, while clerics' buffs will help them in any physical combat.  Maybe that's ok?  Maybe we should roll a weaker weapon resistance into shield too?

**Comment by XTremeTennis** [comment]
could have elemental shield that does the elemental resistance, and then a secondary spell (mage armor or shield, whatever) that does physical resistance. 
One appeal I think of having the shield spells be different is having to choose which energy type you are resistant too rather than it being a catchall

**Comment by sansthesis** [comment]
Yes, I'd like there to be some strategy involved, but making 5 extremely situational spells is counter to reducing the overall number of spells.

If there's a way to keep the bloat down with that, we could give "mage armor" to both mages and clerics and have it provide weapon resistance, and then they each get their own elemental protection spells (energy/fire/frost/shocking/acid) and (holy/unholy).

**Comment by XTremeTennis** [comment]
So, is your concern with spell bloat on the balance/code side, or the playability side? If its a concern with there just being too many spells on casters, then we could limit the amount of spells you could possibly get, force specialization so that you cant get all of them.
Also, I'm iffy on holy/unholy because it puts neutral casters in a weird spot, I don't know if its possible to just have one damage type thats 'divine' or something and then have it be pure alignment, so that a good cleric/paladin can't do nearly as much damage to other good aligned things, but can do full damage to neutral/evil

**Comment by sansthesis** [comment]
Having too many spells in general leads to balance issues, yes, but there's tons of spells out there that are so situational that they're not useful -- faerie fog, floating disc, create food/water, ventriloquate, call lightning, detect poison, dispel good/neutral/evil, holy word, all the way down the list.  It's too much to keep in mind and too much to waste practices or spamming time on.  The cleric spell list looks much more compelling than the mage one atm due to clerics having a tight focus and few situational combat spells.

Holy and unholy are two other damage types in the codebase and have no bearing upon being just two more elemental types.  I don't care too much if they get merged into one or not.

**Comment by sansthesis** [comment]
Mage spells (35 atm):

| Category | Energy | Fire | Frost | Shocking | Acid |
| --- | --- | --- | --- | --- | --- |
| Single Target | Magic Missile | Burning Hands | Ray of Frost | Lightning Bolt | Acid Blast |
| AOE | Prismatic Spray | Fireball | Cone of Cold | Chain Lightning | Horrid Wilting |
| Shield | Energy Shield | Flameshield | Frostshield | Shockshield | Earthshield |

| Debuffs |
| --- |
| Sleep |
| Slow |
| Embrittle |
| Confusion |
| Silence |
| Dispel Magic |

| Buffs |
| --- |
| Haste |
| Mage Armor |
| Invisibility |
| Fly |
| Detect Invis |

| Utility |
| --- |
| Portal |
| Summon |
| Cancellation |
| Scrye |
| Enchant Weapon |
| Enchant Armor |
| Find Familiar |
| Charm Person |
| Teleport |

**Comment by sansthesis** [comment]
Cleric spells (30 atm):

| Target | Spell | Notes |
| --- | --- | --- |
| Single | Harm | Pure damage to single target, maybe magic in nature? |
| Single | Purge | Damage affected by alignment, can do more than harm to outside of align but less inside align? |
| AoE | Wrath | AoE alignment based spell |

Healing:

| Spell | Note |
| --- | --- |
| Cure | Obtained at level 1, heals in a range that gets bigger the higher in level you are |
| Heal | Obtained later, costs more mana than cure but heals an assigned amount every time |
| Refresh | Heals moves |
| Cure Poison | Cures poison |
| Cure Blindness | Cures blindness |
| Remove Curse | Removes curse |

Buffs:

| Spell | Notes |
| --- | --- |
| Bless | Bonus to hitroll and maybe saves |
| Mage Armor | bonus to Ac or DR or both |
| Sanctuary | Damage reduction |
| Divine Blessing | I think an alignment based buff spell would be cool, where it works different based on what the casters align is |
| Protection | less damage from those outside alignment, then we can remove protection X alignment |
| Invisibility | Turns target invisible |
| Fly | Makes target fly |
| Detect Invis | Detects invisible things |
| Giant Strength | Increases strength |

Debuffs:

| Spell | Notes |
| --- | --- |
| Calm | makes target unable to attack |
| Blindness | blinds someone |
| Weaken | lowers strength/damage roll |
| Curse | lowers hitroll |
| Poison | damage over time |
| Dispel Magic | Removes buffs |

| Utility |
| --- |
| Portal |
| Summon |
| Cancellation |
| Scrye |
| Charm Person |
| Teleport |

**Comment by XTremeTennis** [comment]
So it looks like the real question is what to do with the shield spells for mage. Also, do we want giant strength or free action?

**Comment by sansthesis** [comment]
I missed giant strength, but free action doesn't make sense without the spells that necessitate it, like entangle, etc.

**Comment by XTremeTennis** [comment]
another thing is cancel, I think we could make buff spells willingly dropped, but I think you can try to cancel maladicts, right?

**Comment by jasonrose** [comment]
Yes, I left it in to remove maledictions.

**Comment by XTremeTennis** [comment]
29 cleric spells, 35 for mages, that seems like much much cleaner lists

**Comment by sansthesis** [comment]
We're up to 30 cleric with giant strength.

**Comment by XTremeTennis** [comment]
given that clerics will have an additional combat skill or two, it probably balances out in the end
